tools/reviewer/webapp/src/app/dashboard/diff/diff-discussion/thread-state.service.ts

```diff
@@ -0,0 +1,69 @@
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs';
+
+import { Thread } from '@/core/proto';
+import { CommentExpandedMap } from '@/shared/thread';
+
+interface ThreadExpandedMap {
+  [id: string]: CommentExpandedMap;
+}
+
+interface FreezeMap {
+  [id: string]: boolean;
+}
+
+@Injectable()
+export class ThreadStateService {
```

Comment:
Can you add some documentation on what type of state we need to save? What is freeze mode?

Reply:
There's explanation what freeze mode is inside thread component file. I don't know what do to best. Should I copy it everywhere, where freeze mode is used?
https://github.com/google/startup-os/blob/WD35/tools/reviewer/webapp/src/app/shared/thread/thread.component.ts#L14

Reply:
Ok, that's fine.